packages/typespec-vscode/src/vscode-cmd/create-tsp-project.ts

```diff
@@ -0,0 +1,664 @@
+    const label =
+      (x.template.title ?? x.name) +
+      ` (min compiler ver: ${x.template.compilerVersion ? x.template.compilerVersion : "-not specified-"})`;
+    return {
+      label,
+      detail: x.template.description,
+      kind: vscode.QuickPickItemKind.Default,
+      info: x,
+    };
+  };
+  // Templates from compiler should always be on the top
+  templateInfoMap.get(COMPILER_CORE_TEMPLATES)?.forEach((x) => {
+    templatePickupItems.push(toPickupItems(x));
+  });
+  for (const key of templateInfoMap.keys()) {
+    if (key === COMPILER_CORE_TEMPLATES) {
+      continue;
+    }
+    const temps = [];
+    for (const info of templateInfoMap.get(key) ?? []) {
+      if (!info || !info.template) {
+        logger.warning(`Template ${info.name} in ${key} is empty. Skip it.`);
+        continue;
+      }
+      temps.push(toPickupItems(info));
+    }
+    if (temps.length > 0) {
+      templatePickupItems.push({
+        label: key,
+        kind: vscode.QuickPickItemKind.Separator,
+      });
+      templatePickupItems.push(...temps);
+    }
+  }
+  templatePickupItems.push({
+    label: "Settings",
+    kind: vscode.QuickPickItemKind.Separator,
+    info: undefined,
+  });
+  const configureSettingsItem: TemplateQuickPickItem = {
+    label: "Configure TypeSpec Project Templates",
+    kind: vscode.QuickPickItemKind.Default,
+    info: undefined,
+    buttons: [
+      {
+        iconPath: new vscode.ThemeIcon("settings-gear"),
+        tooltip: "Configure TypeSpec Project Templates",
+      },
+    ],
+  };
+  templatePickupItems.push(configureSettingsItem);
+  const quickPickup = vscode.window.createQuickPick<TemplateQuickPickItem>();
+  quickPickup.items = templatePickupItems;
+  quickPickup.canSelectMany = false;
+  quickPickup.ignoreFocusOut = true;
+  quickPickup.title = "Please select a template";
+  quickPickup.placeholder = "Please select a template";
+  const gotoConfigSettings = () => {
+    logger.info("User select to open settings to configure TypeSpec Project Templates");
+    quickPickup.hide();
+    vscode.commands.executeCommand("workbench.action.openSettings", SettingName.InitTemplatesUrls);
+  };
+  quickPickup.onDidTriggerItemButton((event) => {
+    if (event.item === configureSettingsItem) {
+      gotoConfigSettings();
+    }
+  });
+  const selectionPromise = new Promise<TemplateQuickPickItem | undefined>((resolve) => {
+    quickPickup.onDidAccept(() => {
+      const selectedItem = quickPickup.selectedItems[0];
+      resolve(selectedItem);
+      quickPickup.hide();
+    });
+    quickPickup.onDidHide(() => {
+      resolve(undefined);
+      quickPickup.dispose();
+    });
+  });
+  quickPickup.show();
+
+  const selected = await selectionPromise;
+  if (configureSettingsItem === selected) {
+    gotoConfigSettings();
+    return undefined;
+  }
+  return selected?.info;
+}
+
+async function IsCompilerSupport(client: TspLanguageClient): Promise<boolean> {
```

Comment:
Style: camelcase function names
```suggestion
async function isCompilerSupport(client: TspLanguageClient): Promise<boolean> {
```

Reply:
updated. good catch!